### PR TITLE
Added data upload and Quiz interface feature

### PR DESCRIPTION
**Pull request by lavish123409**
## Purpose ✨

The Question adding feature is now integrated with the database and the quiz interface on the client side is working correctly.

## Details 📝

- The user can now add the quiz to the DB
- Added the timer for the question
- User can now select any option
- When the user submits the answer, he would be shown the correct verdict for it.

## Dependencies 🔗

Axios - it is used to make API calls to the server

## Future Improvements 🔭

Leaderboard.jsx - would be updated to show real scores to user fetched from DB and also write to it

## Mentions 👀

@Gopal-Dahale @KShivendu 

## Developer's checklist 📃
- [x] Followed the [coding guidelines](https://google.github.io/styleguide/jsguide.html).
- [x] Properly commented the code.
- [ ] No print statements in the code. (***Some console.log statements are there for debugging***)
- [x] All the functionalities are working properly.
- [x] Changes made are not breaking any other part of the project.
- [ ] No UI/UX issues are present. (***The timer is not stopping instantaneously on submitting the answer.***)
- [x] Followed the PR guidelines

### CODE REVIEW

File: client/src/Containers/Addquestion.jsx [hunk 1]
```diff
@@ -1,49 +1,59 @@
...
+  const sendData = () => {
+    const question = document.getElementById('question-input').value;
+    const option1 = document.getElementById('option1-input').value;
+    const option2 = document.getElementById('option2-input').value;
+    const option3 = document.getElementById('option3-input').value;
+    const option4 = document.getElementById('option4-input').value;
+    const correctAnswer = document.getElementById('correct-answer').value;
+    const timeGiven = document.getElementById('time-given').value;
+  
+    const obj = {
```
Comment (Gopal-Dahale) [inline client/src/Containers/Addquestion.jsx:27]: Given a useful variable name

File: client/src/Containers/Addquestion.jsx [hunk 2]
```diff
@@ -148,7 +173,7 @@ const Addquestion = () => {
 
             <Button
               variant="contained"
-              href="/addtitle"
+              onClick={ () => setOpen(true)}
```
Comment (Gopal-Dahale) [inline client/src/Containers/Addquestion.jsx:176]: It's not formatted correctly. Aren't you using `prettier` to format?

File: client/src/Containers/Addtitle.jsx [hunk 3]
```diff
@@ -1,59 +1,119 @@
...
 
 
-const Addquestion = () => {
-  const classes = useStyles();
+const Addtitle = ({ open , setOpen , data , setData }) => {
+  const [quizLink, setQuizLink] = useState('');
+  const [openQuizLink, setOpenQuizLink] = useState(false);
+  
+
+  function sendToDB()
```
Comment (Gopal-Dahale) [inline client/src/Containers/Addtitle.jsx:13]: Use arrow function.

File: client/src/Containers/Addtitle.jsx [hunk 4]
```diff
@@ -1,59 +1,119 @@
...
+  {
+    const title = document.getElementById('title-input').value; // Get the title
+    setData( dataObj => {
+      const obj = dataObj;
```
Comment (Gopal-Dahale) [inline client/src/Containers/Addtitle.jsx:17]: proper variable name.

File: client/src/Containers/Addtitle.jsx [hunk 5]
```diff
@@ -1,59 +1,119 @@
...
+      obj.title = title;
+      
+      return obj;
+    }); // Adding the title to the object to be sent
+
+    axios.post( 'http://localhost:5000/addquiz' , data)
```
Comment (Gopal-Dahale) [inline client/src/Containers/Addtitle.jsx:23]: Explain the axios post with proper comments like what this post request is doing?

File: client/src/Containers/Addtitle.jsx [hunk 6]
```diff
@@ -1,59 +1,119 @@
...
   );
 };
 
-export default Addquestion;
+Addtitle.propTypes = {
+  open : PropTypes.bool.isRequired,
+  setOpen : PropTypes.func.isRequired,
+  data : PropTypes.shape.isRequired,
+  setData : PropTypes.func.isRequired
+}
```
Comment (Gopal-Dahale) [inline client/src/Containers/Addtitle.jsx:117]: It's good that you have added PropTypes. 👍

File: client/src/Containers/Home.jsx [hunk 7]
```diff
@@ -4,7 +4,9 @@ import useStyles from './Styles/HomeStyles';
 
 const goToQuiz = () => {
   const quizCode = document.getElementById('quiz-code').value; // This value can be used to go to the quiz page
-  console.log('Quiz code : ', quizCode);
+  // console.log('Quiz code : ', quizCode);
```
Comment (Gopal-Dahale) [inline client/src/Containers/Home.jsx:7]: Remove commented console statements.

File: client/src/Containers/QuizArea.jsx [hunk 8]
```diff
@@ -1,17 +1,214 @@
...
+  const selectdButtonRef = useRef(null);
+  const correctOptionRef = useRef(null);
+  const showAnswerRef = useRef(false);
+
+  let interval;
+
+
+
+  // This function is used to color the option button when it is selected
+  function colorButton(optionID , bgcolor , color)
```
Comment (Gopal-Dahale) [inline client/src/Containers/QuizArea.jsx:33]: Use arrow function.

File: client/src/Containers/QuizArea.jsx [hunk 9]
```diff
@@ -1,17 +1,214 @@
...
+    }
+    else // if the option selected by user is incorrect
+    {
+        colorButton(correctOptionRef.current , '#00e676' , 'white'); // show the correct option in green
+        colorButton(selectdButtonRef.current , '#f50057' , 'white'); // show the selected incorrect option in red
+    }
+  }
+
+  
+  useEffect(() => {
```
Comment (Gopal-Dahale) [inline client/src/Containers/QuizArea.jsx:95]: Above each useEffect make sure to add comments mentioning the purpose of useEffect.

File: client/src/Containers/QuizArea.jsx [hunk 10]
```diff
@@ -1,17 +1,214 @@
...
+        axios.get(`http://localhost:5000/quiz/${quizid}`)
```
Comment (Gopal-Dahale) [inline client/src/Containers/QuizArea.jsx:96]: Explain this get request with comments.

File: client/src/Containers/QuizArea.jsx [hunk 11]
```diff
@@ -1,17 +1,214 @@
...
+        // return () => {
+        //     clearInterval(interval);
+        // }
+  }, []);
+
+  useEffect(() => {
+
+    // This use effect would run whenever the index value of question changes
+    // It would change the question, its corresponding options on quiz interface
+    // and would also reset the timer
```
Comment (Gopal-Dahale) [inline client/src/Containers/QuizArea.jsx:116]: Put it above useEffect in this (/** */) format.

File: client/src/Containers/QuizArea.jsx [hunk 12]
```diff
@@ -1,17 +1,214 @@
...
+    {
+        timeRef.current = quizData.questions[index].time_given; // reset the timer value to the time allotted for the question
+        setTime(timeRef.current); // set the time value
+        setTimeGivenforQ(timeRef.current); // set the original time allotted for the given question
+    }
+
+    if(interval) // [ NOT WORKING ] if timer is running then stop it
+    clearInterval(interval);
+      
+    // -------------------------------Timer------------------------------------------------
```
Comment (Gopal-Dahale) [inline client/src/Containers/QuizArea.jsx:139]: Use this (/*************** Timer ****************/) style of commenting for a block of code.

**Review by Gopal-Dahale — CHANGES_REQUESTED** [review]
@lavish123409 I have left some comments.

**Comment by Gopal-Dahale** [comment]
In the purpose section, write 1-2 lines about the actual purpose and write other things in the details section (currently whatever you have written in the purpose section should be in the details section). Also, change the name of the pull request from `Baasic` to some useful name.

File: client/src/Containers/Addquestion.jsx (hunk 2 above)
Comment (lavish123409) [reply client/src/Containers/Addquestion.jsx:176]: How it should be formatted? `prettier` is not giving any error.

File: client/src/Containers/Addtitle.jsx (hunk 6 above)
Comment (lavish123409) [reply client/src/Containers/Addtitle.jsx:120]: Thanks😊

**Comment by lavish123409** [comment]
I have tried to do as many as requested changes. Kindly review it once again @Gopal-Dahale @KShivendu

**Comment by Gopal-Dahale** [comment]
LGTM :+1:

**Pull request merged**